great_expectations/experimental/datasources/pandas_azure_blob_storage_datasource.py

```diff
@@ -0,0 +1,339 @@
+            data_connector=data_connector,
+            test_connection_error_message=test_connection_error_message,
+        )
+
+    def add_json_asset(
+        self,
+        name: str,
+        batching_regex: Union[re.Pattern, str],
+        container: str,
+        name_starts_with: str = "",
+        delimiter: str = "/",
+        order_by: Optional[BatchSortersDefinition] = None,
+        **kwargs,  # TODO: update signature to have specific keys & types
+    ) -> JSONAsset:  # type: ignore[valid-type]
+        """Adds a JSON DataAsst to the present "PandasAzureBlobStorageDatasource" object.
+
+        Args:
+            name: The name of the JSON asset
+            batching_regex: regex pattern that matches csv filenames that is used to label the batches
+            container: container name for Microsoft Azure Blob Storage
+            name_starts_with: Microsoft Azure Blob Storage object name prefix
+            delimiter: Microsoft Azure Blob Storage object name delimiter
+            order_by: sorting directive via either list[BatchSorter] or "{+|-}key" syntax: +/- (a/de)scending; + default
+            kwargs: Extra keyword arguments should correspond to ``pandas.read_csv`` keyword args
+        """
+        batching_regex_pattern: re.Pattern = self.parse_batching_regex_string(
+            batching_regex=batching_regex
+        )
+        order_by_sorters: list[BatchSorter] = self.parse_order_by_sorters(
+            order_by=order_by
+        )
+
+        asset = JSONAsset(
+            name=name,
+            batching_regex=batching_regex_pattern,
+            order_by=order_by_sorters,
+            **kwargs,
+        )
+
+        data_connector: DataConnector = AzureBlobStorageDataConnector(
+            datasource_name=self.name,
+            data_asset_name=name,
+            batching_regex=batching_regex_pattern,
+            azure_client=self._azure_client,
+            account_name=self._account_name,
+            container=container,
+            name_starts_with=name_starts_with,
+            delimiter=delimiter,
+            file_path_template_map_fn=AzureUrl.AZURE_BLOB_STORAGE_HTTPS_URL_TEMPLATE.format,
+        )
+        test_connection_error_message: str = f"""No file belonging to account "{self._account_name}" in container "{container}" with prefix "{name_starts_with}" matched regular expressions pattern "{batching_regex_pattern.pattern}" using delimiter "{delimiter}" for DataAsset "{name}"."""
+        return self.add_asset(
+            asset=asset,
+            data_connector=data_connector,
+            test_connection_error_message=test_connection_error_message,
+        )
+
+    def add_parquet_asset(
+        self,
+        name: str,
+        batching_regex: Union[re.Pattern, str],
+        container: str,
+        name_starts_with: str = "",
+        delimiter: str = "/",
+        order_by: Optional[BatchSortersDefinition] = None,
+        **kwargs,  # TODO: update signature to have specific keys & types
```

Comment:
I thought this TODO was resolved in the filesystem code?

Reply:
@donaldheppner I am not sure -- perhaps @Kilo59 knows; I have been leaving them as is.  Thanks.

Reply:
It is and you've also copied the signature modifying code below.
You can remove this TODO comment.

Reply:
@Kilo59 I committed a change where TODOs are removed -- could you please verify that I did not overdo it?  Thank you.